migrate/repo_checks.py

```diff
@@ -0,0 +1,364 @@
+        self.cla_team = "cla-checker"
+
+        self.has_a_branch_protection_rule = False
+        self.branch_protection_has_required_checks = False
+        self.required_checks_has_cla_required = False
+        self.team_setup_correctly = False
+
+    def check(self):
+        is_required_check = self._check_cla_is_required_check()
+        repo_on_required_team = self._check_cla_team_has_write_access()
+
+        value = is_required_check[0] and repo_on_required_team[0]
+        reason = f"{is_required_check[1]} {repo_on_required_team[1]}"
+        return (value, reason)
+
+    def _check_cla_is_required_check(self):
+        repo = self.api.repos.get(self.org_name, self.repo_name)
+        default_branch = repo.default_branch
+        # Branch protection rule might not exist.
+        try:
+            branch_protection = self.api.repos.get_branch_protection(
+                self.org_name, self.repo_name, default_branch
+            )
+            self.has_a_branch_protection_rule = True
+        except HTTP404NotFoundError as e:
+            return (False, "No branch protection rule.")
+
+        if "required_status_checks" not in branch_protection:
+            return (False, "No required status checks in place.")
+        self.branch_protection_has_required_checks = True
+
+        # We don't need to check the `contexts` list because, github mirrors
+        # all existing checks in `contexts` into the `checks` data.  The `contexts`
+        # data is deprecated and will not be available in the future.
+        contexts = [
+            check["context"]
+            for check in branch_protection.required_status_checks.checks
+        ]
+        if "openedx/cla" not in contexts:
+            return (False, "CLA Check is not a required check.")
+        self.required_checks_has_cla_required = True
+
+        return (True, "Branch Protection with CLA Check is in Place.")
+
+    def _check_cla_team_has_write_access(self):
+        teams = chain.from_iterable(
+            paged(
+                self.api.repos.list_teams,
+                self.org_name,
+                self.repo_name,
+                per_page=100,
+            )
+        )
+
+        team_permissions = {team.slug: team.permission for team in teams}
+        if self.cla_team not in team_permissions:
+            return (False, f"'{self.cla_team}' team not listed on the repo.")
+        # CLA Checker needs write access to push status but it doesn't need anything
+        # higher than that.
+        elif team_permissions[self.cla_team] != "push":
+            return (
+                False,
+                f"'{self.cla_team}' team does not have the correct access. "
+                f"Has {team_permissions[self.cla_team]} instead of push.",
+            )
+        else:
+            self.team_setup_correctly = True
+            return (True, f"'{self.cla_team}' team has 'push' access.")
+
+    def dry_run(self):
```

Comment:
Nit: I think it would make sense to make `dry_run` a member of the class instead of passing it around since it doesn't look like you're calling the `fix` method more than one way in the same run. I can see an argument for leaving it as is for manual testing, it really depends on how you expect it to be used.